arch/x86/core/x86_mmu.c

```diff
@@ -15,10 +15,13 @@
 #include <logging/log.h>
 LOG_MODULE_DECLARE(os);
 
+#define PHYS_RAM_ADDR DT_REG_ADDR(DT_CHOSEN(zephyr_sram))
+#define PHYS_RAM_SIZE DT_REG_SIZE(DT_CHOSEN(zephyr_sram))
+
 /* Despite our use of PAE page tables, we do not (and will never) actually
  * support PAE. Use a 64-bit x86 target if you have that much RAM.
  */
-BUILD_ASSERT(DT_PHYS_RAM_ADDR + (DT_RAM_SIZE * 1024ULL) - 1ULL <=
+BUILD_ASSERT(PHYS_RAM_ADDR + PHYS_RAM_SIZE - 1ULL <=
```

Comment:
This isn't right, PHYS_RAM_SIZE needs to be multiplied by 1024 to get it in bytes not kilobytes.
Consider just incorporating the 1024 coefficient to the PHYS_RAM_SIZE definition, will make this simpler.

Reply:
PHYS_RAM_SIZE is now the size in bytes.

Reply:
DT_REG_SIZE() gives us the size in bytes, that I removed the 1024 scaling.

I wonder if there's a common x86 header to put:
```
#define PHYS_RAM_ADDR DT_REG_ADDR(DT_CHOSEN(zephyr_sram))
#define PHYS_RAM_SIZE DT_REG_SIZE(DT_CHOSEN(zephyr_sram))
```

Reply:
If this is already in bytes you need to get rid of the 1024 multiplication to PHYS_RAM_SIZE in the rest of this file!